syntax = "proto3";
package xml;

service XmlService {
  rpc CsvToXml (CsvToXmlRequest) returns (OperationReply) {}
  rpc XmlToXsd (XmlToXsdRequest) returns (OperationReply) {}
  rpc ValidateXml (ValidateXmlRequest) returns (OperationReply) {}
  rpc ProcessXml (ProcessXmlRequest) returns (OperationReply) {}
  rpc GetCollections (GetCollectionsRequest) returns (GetCollectionsReply) {}
}

message CsvToXmlRequest { string csv_name = 1; }
message XmlToXsdRequest { string xml_name = 1; }
message ValidateXmlRequest { string xml_name = 1; string xsd_name = 2; }
message ProcessXmlRequest { string xml_name = 1; }
message GetCollectionsRequest {}

message OperationReply { bool success = 1; string message = 2; }
message GetCollectionsReply { repeated string collections = 1; }
